components/charts/ChartFigure/ChartFigure.stories.js

```diff
@@ -0,0 +1,197 @@
+import ChartFigure from ".";
+import ChartBase from "../ChartBase";
+import XAxis from "../XAxis";
+import YAxis from "../YAxis";
+import BarData from "../BarData";
+import LineData from "../LineData";
+import ChartLegend from "../Legend";
+import { getLinearScale } from "@/lib/utils";
+
+const meta = {
+  component: ChartFigure,
+  argTypes: {},
+};
+export default meta;
+
+const PrimaryTemplate = ({ ...args }) => {
+  const width = 900;
+  const height = 300;
+  const padding = 40;
+  const xDomain = [0, 24];
+  const tempDomain = [0, 50];
+  const xScale = getLinearScale(xDomain, [padding, width - padding]);
+  const tempScale = getLinearScale(tempDomain, [height - padding, 0]);
+  const legends = [
+    { title: "Temperature", type: "line", id: "legendTemperature" },
+    { title: "Dew Point", type: "bar", id: "legendDewPoint" },
+  ];
+
+  const dewPointData = [
+    6, 6, 5, 5, 4, 4, 3, 2.5, 2, 8, 11, 15, 19, 22, 24, 23, 19, 14, 12, 9, 8, 7,
+    6.5, 6,
+  ];
+
+  const tempData = [
+    25, 23, 20, 18, 17, 15, 13, 10, 8, 9, 11, 15, 18, 28, 34, 32, 27, 23, 21,
+    19, 15, 12, 14, 22,
+  ];
+
+  return (
+    <ChartFigure {...args} legend={<ChartLegend legends={legends} />}>
+      <ChartBase {...{ width, height }}>
+        <XAxis
+          {...{
+            xDomain,
+            xScale,
+            y: height - padding,
+            ticks: 24,
+            labelFormatter: (value) => {
```

Comment:
label formatters are a powerful way to adjust labels either using unit formatting or dynamically (see the "NOW" label below)